sdk/core/core/test/cmocka/test_az_http.c

```diff
@@ -357,6 +358,82 @@ enable_precondition_check_tests()
       az_http_request_append_header(&hrb, AZ_SPAN_FROM_STR(" \t\r"), AZ_SPAN_NULL));
 }
 
+static void test_http_request_header_validation(void** state)
+{
+  (void)state;
+  {
+    uint8_t header_buf[(2 * sizeof(az_pair))];
+    memset(header_buf, 0, sizeof(header_buf));
+
+    az_span url_span = AZ_SPAN_FROM_STR("some.url.com");
+    az_span header_span = AZ_SPAN_FROM_BUFFER(header_buf);
+    _az_http_request hrb;
+
+    TEST_EXPECT_SUCCESS(az_http_request_init(
+        &hrb,
+        &az_context_app,
+        az_http_method_get(),
+        url_span,
+        az_span_size(url_span),
+        header_span,
+        AZ_SPAN_FROM_STR("body")));
+
+    assert_precondition_checked(az_http_request_append_header(
+        &hrb, AZ_SPAN_FROM_STR("(headerName)"), hrb_header_content_type_token));
+
+    // make sure about header was not added
+    assert_int_equal(az_http_request_headers_count(&hrb), 0);
+    _az_http_request_headers headers = hrb._internal.headers;
+    size_t size = (size_t)az_span_size(headers);
+    assert_memory_equal(header_buf, headers._internal.ptr, size);
+  }
+}
+
+static void test_http_request_header_validation_above_127(void** state)
+{
+  (void)state;
+  {
+    uint8_t header_buf[(2 * sizeof(az_pair))];
+    memset(header_buf, 0, sizeof(header_buf));
+
+    az_span url_span = AZ_SPAN_FROM_STR("some.url.com");
+    az_span header_span = AZ_SPAN_FROM_BUFFER(header_buf);
+    _az_http_request hrb;
+
+    TEST_EXPECT_SUCCESS(az_http_request_init(
+        &hrb,
+        &az_context_app,
+        az_http_method_get(),
+        url_span,
+        az_span_size(url_span),
+        header_span,
+        AZ_SPAN_FROM_STR("body")));
+
+    uint8_t c[1] = { 255 };
+    az_span header_name = AZ_SPAN_FROM_BUFFER(c);
+    assert_precondition_checked(
+        az_http_request_append_header(&hrb, header_name, hrb_header_content_type_token));
+
+    // make sure about header was not added
+    assert_int_equal(az_http_request_headers_count(&hrb), 0);
+    _az_http_request_headers headers = hrb._internal.headers;
+    size_t size = (size_t)az_span_size(headers);
+    assert_memory_equal(header_buf, headers._internal.ptr, size);
+  }
+}
+
+static void test_http_request_header_validation_range(void** state)
+{
+  (void)state;
+  {
+    // just make sure compiler will set anything greater than 127 to zero
+    for (uint8_t value = 127; value > 127; value++) // break on unsigned back to zero
```

Comment:
Why rely on overflowing?
Change the boundary condition to be 256.
```suggestion
    for (uint8_t value = 127; value < 256; value++)
```

Reply:
using `< 256` is an infinite loop.  Compiler throws a 'always true condition` con compilation time.
uint8_t range is 0-255.

Reply:
Oh, lol, right. Good point. Use `int32_t` then and cast?

Reply:
no need to cast if we can directly use uint8_t.  And this is at the end, just a test, so, would be fine either way. 
I would just keep it with no casting